hazelcast/src/main/java/com/hazelcast/map/impl/operation/EntryBackupOperation.java

```diff
@@ -93,6 +122,7 @@ private boolean entryRemovedBackup(Map.Entry entry) {
         final Object value = entry.getValue();
         if (value == null) {
             recordStore.removeBackup(dataKey);
+            publishWanReplicationEvent(EntryEventType.REMOVED);
```

Comment:
The problem with this approach is that again the wan replication is sprinkled over the code base. It is now too late to fix it; but we need to come up with a better solution. Wan replication should be integrated directly into the operation service. E.g. as part of a custom OperationBackupHandler.